recipe/meta.yaml

```diff
@@ -1,113 +1,85 @@
+  patches:
+    - patches/0001-backend_torch.patch
 
 build:
-  noarch: python
   number: 0
+  script: 
+    # Remove eventual leftover config (will set the backend)
+    # linux instances do not have this problem because they
+    # are fresh every time.
+    # See: https://github.com/keras-team/keras/blob/v3.0.5/keras/backend/config.py#L199
+    - rm -f ~/.keras/keras.json                   # [osx]
+    - del %UserProfile%\.keras\keras.json 2> nul  # [win]
+    - {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-build-isolation -v
+  # skipping s390x since dm-tree and jax are not available
+  skip: true # [py<39 or s390x]
 
-outputs:
-  - name: keras-base
-    script: install_base.sh  # [not win]
-    script: install_base.bat  # [win]
-    build:
-      noarch: python
-    requirements:
-      host:
-        - python
-        - pip
-        - setuptools
-        - wheel
-      run:
-        - python >=3.6
-        - numpy >=1.9.1
-        - scipy >=0.14
-        - pyyaml
-        - h5py
-        - tensorflow {{ version }}  
-      # add run_constrained to prevent installation of mismatched keras and
-      # keras-base packages
-      run_constrained:
-        - keras {{ version }}
-    test:
-      requires:
-        - pip
-      commands:
-        - pip check
-      imports:
-        - keras
-        - keras.backend
-        - keras.datasets
-        - keras.engine
-        - keras.layers
-        - keras.preprocessing
-        - keras.utils
-        - keras.wrappers
- #some of the imports are commented out during testing to avoid circular dependencies
-  - name: keras
-    script: install_base.sh  # [not win]
-    script: install_base.bat  # [win]
-    build:
-      noarch: python
-    requirements:
-      host:
-        - python
-        - pip
-        - setuptools
-        - wheel
-      run:
-        - python >=3.6
-      run_constrained:
-        - tensorflow {{ version }}
-    test:
-      requires:
-        - pip
-      commands:
-        - pip check
-    about:
-      home: https://github.com/fchollet/keras
-      license: MIT
-      license_family: MIT
-      license_file: LICENSE
-      summary: Deep Learning Library for Theano and TensorFlow
-      description: |
-        Keras is a minimalist, highly modular neural networks library
-        written in Python and capable on running on top of either
-        TensorFlow or Theano. It was developed with a focus on enabling
-        fast experimentation. Being able to go from idea to result with
-        the least possible delay is key to doing good research.
-      doc_url: http://keras.io
-      dev_url: https://github.com/fchollet/keras
+requirements:
+  build:
+    - patch    #  [not win]
+    - m2-patch #  [win]
+  host:
+    - python
+    - pip
+    - wheel
+    - setuptools
+  run:
+    - python
+    - absl-py
+    - numpy
+    - rich
+    - namex
+    - h5py
+    - ml_dtypes
+    - packaging
```

Comment:
```suggestion
```
Seems `packaging` isn't required at runtime, see https://github.com/keras-team/keras/blob/v3.0.5/setup.py#L41-L48
It was added to the `master` branch and versions **>=3.3.0**

Reply:
removed, thank you

Reply:
Well, it's really needed:
```
  File "/croot/keras_1716277111998/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.12/site-packages/keras/utils/torch_utils.py", line 3, in <module>

    from packaging.version import parse
ModuleNotFoundError: No module named 'packaging'
```

Reply:
I need to restore it: `packaging` is used in:
https://github.com/keras-team/keras/blob/v3.0.5/keras/utils/torch_utils.py#L3

Reply:
See log at https://cloud.prefect.io/anaconda/flow-run/c8a23c40-6ec4-4650-8952-ec8f5c4e9e36

Reply:
yup, saw it now ehehe

Reply:
restored, added a comment in the recipe